agent/taskresource/credentialspec/credentialspec_windows.go

```diff
@@ -455,29 +468,23 @@ func (cs *CredentialSpecResource) writeS3File(writeFunc func(file oswrapper.File
 	if err != nil {
 		return err
 	}
-	defer temp.Close()
 
 	err = writeFunc(temp)
 	if err != nil {
 		return err
 	}
 
-	err = temp.Chmod(os.FileMode(filePerm))
-	if err != nil {
-		return err
-	}
-
-	// Persist the file to disk.
-	err = temp.Sync()
+	err = temp.Close()
 	if err != nil {
+		seelog.Errorf("Error while closing the handle to file:%v", temp.Name())
 		return err
 	}
 
 	err = cs.os.Rename(temp.Name(), filePath)
 	if err != nil {
+		seelog.Errorf("Error while renaming the temporary file:%v", temp.Name())
```

Comment:
Please include the error in the message. 
Update this to `"Error while renaming the temporary file %s: %v", temp.Name(), err`

Reply:
This comment is addressed.  For some reason my earlier response was not shown here. Posting again